### DISCUSSION TITLE
Install Doctor: Chezmoi-Based Multi-OS Provisioning System

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [twpayne/chezmoi#3659] [Show and tell]
Hey everyone,

I'm thrilled to unveil a project that's been a labor of love for me over the past few years: an automated system for regularly reprovisioning my desktops called [Install Doctor](https://install.doctor). Initially starting out with Ansible, I eventually migrated to Chezmoi once I grasped the pivotal role dot files play in defining an operating system's state.

After pouring countless hours and a considerable amount of resources into this endeavor, I'm excited to share it with you all. You can explore the project and its documentation portal at https://install.doctor/. I'm presenting it here in the hopes that it serves as a substantial reference point for anyone in search of inspiration. Feedback and pull requests are warmly welcomed.

Presently, the project caters to *nix systems and offers partial support for Qubes deployments. Let's collaborate and make it even better together!

![web-apps](https://github.com/twpayne/chezmoi/assets/59970525/e7cec9be-3cfd-47b4-8c2f-1941b1408e7e)

**GitHub:** https://github.com/megabyte-labs/install.doctor
**Home:** https://install.doctor

Please consider leaving feedback here or in the projects various social web properties.

**Comment by twpayne** [comment]
Thank you, this looks amazing! And thank you for your kind words about chezmoi and its community in the documentation :)

**Comment by Lockszmith-GH** [comment]
Awesome project, I'm going to carve out some time to check it out in more detial.
Would this project be suitable for a layered approach of User and Team configuration?
I've done something like this with chezmoi (see [this post](https://github.com/twpayne/chezmoi/discussions/2574)), and I wonder if I can shift that configuration to use your infrastructure?

**Reply by ProfessorManhattan** [reply]
I need to investigate further, but essentially, it's akin to executing Chezmoi twice, with shared details taking precedence, correct? Essentially, this project includes a kickstart script to streamline the process as much as possible and then executes everything with Chezmoi once. I'd be thrilled to collaborate with you. Implementing user/team settings would be a valuable addition.

At the moment, I'm focused on optimizing everything for asynchronous operations and exploring the integration of JuiceFS for application settings folders. The overarching aim of the project is to facilitate frequent and rapid reformatting and restoration.

**Comment by twitchel** [comment]
This looks fantastic. Decided to give this a run on a clean ubuntu VM and its failing on Homebrew install :( Any tips before i start debugging?
![image](https://github.com/user-attachments/assets/840edb2c-b358-4642-b6f6-b250a05871e7)

**Reply by Lockszmith-GH** [reply]
From your screenshot it's not even clear what you tried doing.

If this has somehting to do with Install Doctor, you should reach out on the [Install Doctor](https://github.com/megabyte-labs/install.doctor?tab=readme-ov-file#install-doctor-the-desktop-provisioning-system) repo, we have zero expereince with that here.

**Reply by twitchel** [reply]
Ah, fair point, definitely not anything chezmoi related. Only replied here because i was following along on this showcase post.

I'll chat to the creator
